File: cpp/15.反转链表.cpp
/*
struct ListNode {
	int val;
	struct ListNode *next;
	ListNode(int x) :
			val(x), next(NULL) {
	}
};*/
class Solution {
public:
    ListNode* ReverseList(ListNode* pHead) {
        if(pHead == nullptr || pHead -> next == nullptr)
            return pHead;
        auto pre = ReverseList(pHead -> next);
        pHead -> next -> next = pHead;
        pHead -> next = nullptr;
        return pre;
    }
};

File: java/15.反转链表.java
/*
public class ListNode {
    int val;
    ListNode next = null;

    ListNode(int val) {
        this.val = val;
    }
}*/
public class Solution {
    public ListNode ReverseList(ListNode head) {
        if (head == null || head.next == null)
            return head;
        ListNode pre = ReverseList(head.next);
        head.next.next = head;
        head.next = null;
        return pre;
    }
}
